cmd/configschema/configschema/resolver.go

```diff
@@ -54,5 +56,62 @@ func NewDirResolver(srcRoot string, moduleName string) DirResolver {
 // PackageDir accepts a Type and returns its package dir.
 func (dr DirResolver) PackageDir(t reflect.Type) string {
 	pkg := strings.TrimPrefix(t.PkgPath(), dr.ModuleName+"/")
+	dir := localPackageDir(dr, pkg)
+	if _, err := os.ReadDir(dir); err != nil {
+		dir = externalPackageDir(dr, pkg)
+	}
+	return dir
+}
+
+// localPackageDir returns the path to a local package.
+func localPackageDir(dr DirResolver, pkg string) string {
 	return path.Join(dr.SrcRoot, pkg)
 }
+
+// externalPackageDir returns the path to an external package.
+func externalPackageDir(dr DirResolver, pkg string) string {
+	line := grepMod(path.Join(dr.SrcRoot, "go.mod"), pkg)
+	v := extractVersion(line)
+	goPath := os.Getenv("GOPATH")
+	if goPath == "" {
+		goPath = build.Default.GOPATH
+	}
+	pkgPath := buildExternalPath(goPath, pkg, v)
+	if _, err := os.ReadDir(pkgPath); err != nil {
+		panic(err)
```

Comment:
Yes I can, out of curiosity what would be the difference in what we are doing? Would it be to reduce the amount of error checking we do?

Reply:
Well not reduce the amount of error checking but make it so that instead of constantly for errors inside the internal function just have the external caller just do one error check for the entire internal function?

Reply:
Yeah, the idea is to let callers decide what to do in case of error, either panic or handle the error in some other way.